src/coreclr/jit/gentree.h

```diff
@@ -5758,11 +5758,13 @@ struct GenTreeCall final : public GenTree
         jitstd::vector<InlineCandidateInfo*>* gtInlineCandidateInfoList;
 
         HandleHistogramProfileCandidateInfo* gtHandleHistogramProfileCandidateInfo;
-        LateDevirtualizationInfo*            gtLateDevirtualizationInfo;
+
         CORINFO_GENERIC_HANDLE compileTimeHelperArgumentHandle; // Used to track type handle argument of dynamic helpers
         void*                  gtDirectCallAddress; // Used to pass direct call address between lower and codegen
     };
 
+    LateDevirtualizationInfo* gtLateDevirtualizationInfo; // Always available for user virtual calls
```

Comment:
Have you checked whether this can be merged with some other information of `GenTreeCall` to avoid growing the size of all large nodes?

Reply:
Late devirtualization can happen even without generic context, so we need to always track the `InlineContext` now despite whether it has a generic context or not. 
We used to union the `gtLateDevirtualizationInfo` with `gtHandleHistogramProfileCandidateInfo` together because we didn't save generic context for methods that have class probe, but now we will have to save an `InlineContext` for them anyway so we can also save the generic context for those methods as well.
This happens to enable late devirtualization for tier-1 methods under PGO and this is why it has diffs for PGO cases only.

Reply:
Instead we can remove the member `class InlineContext* gtInlineContext` from `GenTreeCall` now, though it's debug only.

Reply:
Or alternatively, we will have to block all inlining of devirted methods without a generic context in late devirt.

Reply:
For debug it's ok to use extra memory, we are more concerned with release. I suppose we cannot move this into the union above this one because of `gtStubCallStubAddr`?

Reply:
Yeah, unfortunately. Maybe we can merge it with `gtControlExpr`, do we ever morph a call before inlining?

Reply:
I think it's ok to leave it as a separate field for now. Maybe we should consider some better approaches in the future to reducing the size of `GenTreeCall` (e.g. storing all rarely used fields in a separate object that gets allocated on demand).

Reply:
Okay. I just pushed a commit to remove the redundant flag.